staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation_test.go

```diff
@@ -427,6 +427,99 @@ func TestValidateConditions(t *testing.T) {
 	}
 }
 
+func TestLabelSelectorMatchExpression(t *testing.T) {
+	// Success case
+	testCases := []struct {
+		name            string
+		labelSelector   *metav1.LabelSelector
+		wantErrorNumber int
+		validateErrs    func(t *testing.T, errs field.ErrorList)
```

Comment:
IIRC we don't encourage using anonymous function in test table?

supporting this will make content in test table exceed their responsibility. Also, for validating specific errors I think it could be put in test logic.